client/src/main/java/com/netflix/conductor/client/http/jersey/JerseyRequestHandler.java

```diff
@@ -0,0 +1,185 @@
+            return clientResponse.getEntityInputStream();
+        } catch (UniformInterfaceException e) {
+            handleUniformInterfaceException(e, uri);
+        } catch (RuntimeException e) {
+            handleRuntimeException(e, uri);
+        }
+        return null;
+    }
+
+    @Override
+    public InputStream get(URI uri) {
+        ClientResponse clientResponse;
+        try {
+            clientResponse =
+                    client.resource(uri)
+                            .accept(MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN)
+                            .get(ClientResponse.class);
+            if (clientResponse.getStatus() < 300) {
+                return clientResponse.getEntityInputStream();
+            } else {
+                throw new UniformInterfaceException(clientResponse);
+            }
+        } catch (UniformInterfaceException e) {
+            handleUniformInterfaceException(e, uri);
+        } catch (RuntimeException e) {
+            handleRuntimeException(e, uri);
+        }
+
+        return null;
+    }
+
+    private void handleClientHandlerException(ClientHandlerException exception, URI uri) {
+        String errorMessage =
+                String.format(
+                        "Unable to invoke Conductor API with uri: %s, failure to process request or response",
+                        uri);
+        LOGGER.error(errorMessage, exception);
+        throw new RequestHandlerException(errorMessage, exception);
+    }
+
+    private void handleRuntimeException(RuntimeException exception, URI uri) {
+        String errorMessage =
+                String.format(
+                        "Unable to invoke Conductor API with uri: %s, runtime exception occurred",
+                        uri);
+        LOGGER.error(errorMessage, exception);
+        throw new RequestHandlerException(errorMessage, exception);
+    }
+
+    private void handleUniformInterfaceException(UniformInterfaceException exception, URI uri) {
+        ClientResponse clientResponse = exception.getResponse();
+        if (clientResponse == null) {
+            throw new RequestHandlerException(
+                    String.format("Unable to invoke Conductor API with uri: %s", uri));
+        }
+        try {
+            if (clientResponse.getStatus() < 300) {
+                return;
+            }
+            LOGGER.warn(
+                    "Unable to invoke Conductor API with uri: {}, unexpected response from server: statusCode={}",
+                    uri,
+                    clientResponse.getStatus());
+            throw new RequestHandlerException(
+                    clientResponse.getEntityInputStream(), clientResponse.getStatus());
+        } catch (RequestHandlerException e) {
+            throw e;
+        } catch (ClientHandlerException e) {
+            handleClientHandlerException(e, uri);
+        } catch (RuntimeException e) {
+            handleRuntimeException(e, uri);
```

Comment:
where would those exceptions throw?

Reply:
The exception from `RequestHandler` is handled by the methods in `ClientBase`. See [here](https://github.com/Netflix/conductor/pull/3024/files#diff-30a809f00770184cbad33a71b4bb625478d9cd6007c535b765133cbf69512562R104) for an example.

Reply:
ya but in this `handleUniformInterfaceException` method, I don't see where `clientHandlerException` and `runtimeexception` can be throw?

Reply:
I misunderstood the question. The `handleUniformInterfaceException` is a copy of the [same method in ClientBase](https://github.com/Netflix/conductor/blob/main/client/src/main/java/com/netflix/conductor/client/http/ClientBase.java#L337-L369). I intentionally did not change the code since we do not have enough tests to verify the changes.